src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -58,6 +62,54 @@
 
     private SigningKeyResolver signingKeyResolver;
 
+    Claims expectedClaims = new DefaultClaims();
+
+    @Override
+    public JwtParser expectIssuedAt(Date issuedAt) {
+        if (issuedAt != null) {
+            expectedClaims.setIssuedAt(issuedAt);
+        }
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expectIssuer(String issuer) {
+        expect(Claims.ISSUER, issuer);
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expectAudience(String audience) {
+        expect(Claims.AUDIENCE, audience);
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expectSubject(String subject) {
+        expect(Claims.SUBJECT, subject);
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expectId(String id) {
+        expect(Claims.ID, id);
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expect(String claimName, Object value) {
```

Comment:
I think this and the others should be renamed to `require` - it feels a little more direct and authoritative such that a failure will occur if missing.